diff --git core/modules/field_layout/src/FieldLayoutBuilder.php core/modules/field_layout/src/FieldLayoutBuilder.php
index fdbefd5e8d..1fada6a0a9 100644
--- core/modules/field_layout/src/FieldLayoutBuilder.php
+++ core/modules/field_layout/src/FieldLayoutBuilder.php
@@ -5,6 +5,7 @@ namespace Drupal\field_layout;
 use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
 use Drupal\Core\Entity\EntityFieldManagerInterface;
 use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Form\FormStateInterface;
 use Drupal\field_layout\Display\EntityDisplayWithLayoutInterface;
 use Drupal\Core\Layout\LayoutPluginManagerInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
@@ -107,7 +108,12 @@ class FieldLayoutBuilder implements ContainerInjectionInterface {
         // moving the field in the form structure. If a #group is already set,
         // do not overwrite it.
         if (isset($regions[$field['region']]) && !isset($build[$name]['#group'])) {
-          $build[$name]['#group'] = $field['region'];
+          if(!empty($build['#parents']) && count($build['#parents']) > 1){
+            $build[$name]['#group'] = implode('][', array_merge($build['#parents'], ['_field_layout', $field['region']]));
+          }
+          else {
+            $build[$name]['#group'] = $field['region'];
+          }
         }
       }
       // Ensure this will not conflict with any existing array elements by
